### PR TITLE
WIP: feature(kubernetes): enable encryption

### PR DESCRIPTION
**Pull request by dkropachev**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Test run: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo/107/

It works, but there is one problem, scylla-operator was not published since https://github.com/scylladb/scylla-operator/pull/184
and therefore it can't pick it up at this point.
Will be fully operational only after moving to next release of scylla-operator

kubectl get Cluster sct-cluster -n scylla -o yaml:
```
      volumeMounts:
      - mountPath: /etc/scylla/ssl_conf
        name: scylla-server-certificates
      - mountPath: /etc/scylla/ssl_conf/client
        name: scylla-client-certificates
      - mountPath: /etc/scylla/encrypt_conf
        name: scylla-encrypt-files
      volumes:
      - name: scylla-client-certificates
        secret:
          secretName: scylla-client-certificates
      - name: scylla-server-certificates
        secret:
          secretName: scylla-server-certificates
      - name: scylla-encrypt-files
        secret:
          secretName: scylla-encrypt-files

```

kubectl -n scylla get secret
```
NAME                             TYPE                                  DATA   AGE
default-token-vntnp              kubernetes.io/service-account-token   3      16m
sct-cluster-member-token-zcgmv   kubernetes.io/service-account-token   3      16m
scylla-client-certificates       Opaque                                11     16m
scylla-encrypt-files             Opaque                                1      16m
scylla-server-certificates       Opaque                                4      16m
```

**Comment by dkropachev** [comment]
@enaydanov ,
Unfortunatelly scylla-operator does not update statefull set of scylla nodes when we change these fields, so there is only one way to make it work is to make the part of cluster.yaml and deploy all at once.

**Comment by dkropachev** [comment]
@bentsi , @enaydanov , please be inform that it will not work untill new scylla operator is released

**Comment by enaydanov** [comment]
@bentsi @dkropachev I prefer to put this on hold till we merge GKE' PR.

**Comment by dkropachev** [comment]
> @bentsi @dkropachev I prefer to put this on hold till we merge GKE' PR.

Agree

**Comment by bentsi** [comment]
@dkropachev please resolve conflicts

**Comment by dkropachev** [comment]
Verified here:
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo/115/
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-byo-longevity/418/

File: sdcm/k8s_configs/operator.yaml [hunk 1]
```diff
@@ -2447,7 +2447,7 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
-        image: ${SCT_K8S_SCYLLA_OPERATOR_DOCKER_IMAGE}
+        image: scylladb/scylla-operator:v0.2.4
```
Comment (bentsi) [inline sdcm/k8s_configs/operator.yaml:2450]: why do you hard-code the version?

**Comment by bentsi** [comment]
colsing due to decision that we won't support this for Operator GA

**Pull request closed without merging**